controllers/controller/workspacerouting/solvers/basic_solver.go

```diff
@@ -41,19 +41,19 @@ type BasicSolver struct{}
 
 var _ RoutingSolver = (*BasicSolver)(nil)
 
-func (s *BasicSolver) FinalizerRequired(routing *controllerv1alpha1.WorkspaceRouting) bool {
+func (s *BasicSolver) FinalizerRequired(*controllerv1alpha1.WorkspaceRouting) bool {
 	return false
 }
 
-func (s *BasicSolver) Finalize(routing *controllerv1alpha1.WorkspaceRouting) error {
+func (s *BasicSolver) Finalize(*controllerv1alpha1.WorkspaceRouting) error {
 	return nil
 }
 
-func (s *BasicSolver) GetSpecObjects(routing *controllerv1alpha1.WorkspaceRouting, workspaceMeta WorkspaceMetadata) (RoutingObjects, error) {
+func (s *BasicSolver) GetSpecObjects(routing *controllerv1alpha1.WorkspaceRouting, workspaceMeta WorkspaceMetadata, isOpenShift bool) (RoutingObjects, error) {
```

Comment:
since BasicSolver already mention dependency on platform, what do you think if we push isOpenshift into BasicSolver as a field?
```
// Kubernetes: use Ingresses without TLS
// OpenShift: use Routes with TLS enabled
type BasicSolver struct{}
```

Reply:
That's the idea I couldn't think of yesterday as I was doing this. Thanks!